cobalt/build/gn.py

```diff
@@ -96,18 +96,24 @@ def write_build_args(build_args_path, original_lines, dict_settings,
 
 
 def main(out_directory: str, platform: str, build_type: str,
-         overwrite_args: bool, gn_gen_args: List[str]):
+         gn_gen_args: List[str]):
   platform_path = f'cobalt/build/configs/{platform}'
   dst_args_gn_file = os.path.join(out_directory, 'args.gn')
   src_args_gn_file = os.path.join(platform_path, 'args.gn')
   Path(out_directory).mkdir(parents=True, exist_ok=True)
 
-  if overwrite_args or not os.path.exists(dst_args_gn_file):
-    build_args = get_build_args(src_args_gn_file)
+  build_args = get_build_args(src_args_gn_file)
+  if not os.path.exists(dst_args_gn_file):
     write_build_args(dst_args_gn_file, build_args[0], build_args[1], build_type)
   else:
-    print(f'{dst_args_gn_file} already exists.' +
-          ' Running ninja will regenerate build files automatically.')
+    # Overwrite args.gn, copy the stale args.gn file to stale_args.gn
+    stale_dst_args_gn_file = dst_args_gn_file.replace('args', 'stale_args')
```

Comment:
AFAICT this will fail if the stale args file already exists. Perhaps add a date/timestamp to the name?

Reply:
I verified that this overwrites an existing stale file. Lemme know iof there's any other failure modes

Reply:
As long as you've tested it works in a new directory and an existing directory (with an args.gn file) it should be good

Reply:
I actually read the whole description this time and it looks like it will work as expected on linux. Windows will fail but we probably don't need to worry about that?

https://docs.python.org/3/library/os.html#os.rename

Reply:
Switching to https://docs.python.org/3/library/os.html#os.replace would make it work as expected on both. Perhaps better to use it?

Reply:
Yeah let's ignore windows entirely